packages/flutter_tools/lib/src/cache.dart

```diff
@@ -519,10 +519,15 @@ class Cache {
     if (_hasWarnedAboutStorageOverride) {
       return;
     }
+    // Automated testing of Flutter builds using non-released artifacts
+    // need to be able to use the --fatal-warnings flag without failing
+    // due to their custom artifact download location.
+    final bool oldWarnings = _logger.hadWarningOutput;
```

Comment:
The same code is used elsewhere in the file.  It would be better design to  put this functionality in the Logger class, and provide methods "disableFatalWarnings" and "enableFatalWarnings" on that class, or even just add an optional parameter to the .printWarning method, which could be used like "fatal: false".

Reply:
Can we instead use the approach from https://github.com/flutter/flutter/pull/128335/files#diff-1347ece95db905ddb11c469de65dd66b5f7086bb1069a33770e8101bfcfb8f69 ?

Reply:
Yes, that PR is an alternative to this.

Or are use suggesting that we don't add a flag, and unconditionally suppress the fatality of this warning, but do it by passing an optional argument to logger.warning( fatal: false), as in the other PR.

So keep this PR, drop the other one, but add the argument to logger.warning().

Reply:
Ahh ok, I didn't realize they were alternatives. I would prefer the other PR.

Reply:
I have modified the other PR to remove the top level flag.  I am closing this PR in favor of the other one.